# moco_control Settings -----------------------
# Settings for ros_control control loop
moco_hw_interface:
  loop_hz: 100
  cycle_time_error_threshold: 0.01

# Settings for ros_control hardware interface
hardware_interface:
  joints:
    - joint_J1
    - joint_J2
    - joint_J3
    - joint_J4
    - joint_J5
    - joint_J6
  sim_control_mode: 1 # 0: position, 1: velocity

# Publish all joint states ----------------------------------
# Creates the /joint_states topic necessary in ROS
joint_state_controller:
  type: joint_state_controller/JointStateController
  publish_rate: 50

# Joint Trajectory Controller -------------------------------
# For detailed explanations of parameter see http://wiki.ros.org/joint_trajectory_controller
arm_trajectory_controller:
  type: pos_vel_controllers/JointTrajectoryController
  # These joints can likely just be copied from the hardware_interface list above
  joints:
    - joint_J1
    - joint_J2
    - joint_J3
    - joint_J4
    - joint_J5
    - joint_J6
  constraints:
    goal_time: 5.0
    joint_J1:
      trajectory: 0.60
      goal:       0.15
    joint_J2:
      trajectory: 0.60
      goal:       0.15
    joint_J3:
      trajectory: 0.60
      goal:       0.15
    joint_J4:
      trajectory: 0.60
      goal:       0.15
    joint_J5:
      trajectory: 0.60
      goal:       0.15
    joint_J6:
      trajectory: 0.60
      goal:       0.15
  gains:
    joint_J1: {p: 2.0,  i: 0.0, d: 0.01, i_clamp: 1}
    joint_J2: {p: 2.0,  i: 0.0, d: 0.01, i_clamp: 1}
    joint_J3: {p: 2.0,  i: 0.0, d: 0.01, i_clamp: 1}
    joint_J4: {p: 2.0,  i: 0.0, d: 0.01, i_clamp: 1}
    joint_J5: {p: 2.0,  i: 0.0, d: 0.01, i_clamp: 1}
    joint_J6: {p: 2.0,  i: 0.0, d: 0.01, i_clamp: 1}

  state_publish_rate:  50 # Defaults to 50
  action_monitor_rate: 20 # Defaults to 20
  hold_trajectory_duration: 0 # Defaults to 0.5

# Joint Group Position Controller -------------------------------
# Stream individual commands to several joints at a high rate
streaming_position_controller:
  type: position_controllers/JointGroupPositionController
  joints:
    - joint_J1
    - joint_J2
    - joint_J3
    - joint_J4
    - joint_J5
    - joint_J6
  pids:
    joint_J1: {p: 2.0, i: 0.0, d: 0.01, i_clamp: 1}
    joint_J2: {p: 2.0, i: 0.0, d: 0.01, i_clamp: 1}
    joint_J3: {p: 2.0, i: 0.0, d: 0.01, i_clamp: 1}
    joint_J4: {p: 2.0, i: 0.0, d: 0.01, i_clamp: 1}
    joint_J5: {p: 2.0, i: 0.0, d: 0.01, i_clamp: 1}
    joint_J6: {p: 2.0, i: 0.0, d: 0.01, i_clamp: 1}

# Individual Position Controllers ---------------------------------------
# Allows to send individual ROS msg of Float64 to each joint separately
joint_J1_position_controller:
  type: position_controllers/JointPositionController
  joint: joint_J1
  pid: {p: 100.0, i: 0.01, d: 10.0}
joint_J2_position_controller:
  type: position_controllers/JointPositionController
  joint: joint_J2
  pid: {p: 100.0, i: 0.01, d: 10.0}
